packages/webcert/src/feature/careProvider/CareProviderModal.tsx

```diff
@@ -1,15 +1,17 @@
-import { PopUpModal } from '@frontend/common'
+import { PopUpModal, ResourceLinkType } from '@frontend/common'
 import _ from 'lodash'
 import React from 'react'
 import { useSelector } from 'react-redux'
-import { getUser, selectIsLoadingUserStatistics } from '../../store/user/userSelectors'
+import { getUser, getUserResourceLinks, selectIsLoadingUserStatistics } from '../../store/user/userSelectors'
 import { CareProviderModalContent } from './CareProviderModalContent'
 
 const CareProviderModal: React.FC = () => {
   const user = useSelector(getUser)
   const isLoadingUserStatistics = useSelector(selectIsLoadingUserStatistics, _.isEqual)
+  const userLinks = useSelector(getUserResourceLinks)
+  const chooseUnitLink = userLinks?.find((link) => link.type === ResourceLinkType.CHOOSE_UNIT)
 
-  const showCareProviderModal = !!user && !user?.loggedInUnit.unitId
+  const showCareProviderModal = !user?.loggedInUnit.unitId && !!chooseUnitLink
```

Comment:
If you are checking for the resource link shouldn't the logic of checking if the unit has been set in the user object be moved to backend? If the modal should not be shown the resource link should either be missing or disabled.

Reply:
Yes, I agree. I need help with that though. I don't know how to code that in backend.

Reply:
For this PR the logic needs to be removed and tests needs to be written reflecting that the frontend will only check if the resource link is active/available.